api/src/pcapi/core/finance/models.py

```diff
@@ -811,3 +813,96 @@ class PaymentMessage(Base, Model):
     name: str = sqla.Column(sqla.String(50), unique=True, nullable=False)
     checksum: bytes = sqla.Column(sqla.LargeBinary(32), unique=True, nullable=False)
+
+
+##
+# When an incident is `CREATED` it can be either `CANCELLED` and it's the end of this incident. Or it can be `VALIDATED`.
+##
+class IncidentStatus(enum.Enum):
+    CREATED = "created"
+    VALIDATED = "validated"
+    CANCELLED = "cancelled"
+
+
+class IncidentType(enum.Enum):
+    OVERPAYMENT = "overpayment"
+    COMMERCIAL_GESTURE = "commercial gesture"
+    OFFER_PRICE_REGULATION = "offer price regulation"
+    FRAUD = "fraud"
+
+
+class FinanceIncident(Base, Model):
+    id: int = sqla.Column(sqla.BigInteger, primary_key=True, autoincrement=True)
+    kind: IncidentType = sqla.Column(
+        sqla.Enum(IncidentType, native_enum=False, create_contraint=False),
+        nullable=False,
+    )
+    status: IncidentStatus = sqla.Column(
+        sqla.Enum(IncidentStatus, native_enum=False, create_constraint=False),
+        nullable=False,
+        server_default=IncidentStatus.CREATED.value,
+        default=IncidentStatus.CREATED,
+    )
+
+    venueId = sqla.Column(sqla.BigInteger, sqla.ForeignKey("venue.id"), nullable=False)
+    venue: sqla_orm.Mapped["offerers_models.Venue | None"] = sqla_orm.relationship(
+        "Venue", foreign_keys=[venueId], backref="finance_incidents"
+    )
+
+    details: dict | None = sqla.Column(
+        sqla_mutable.MutableDict.as_mutable(sqla_psql.JSONB), nullable=False, default={}, server_default="{}"
+    )
+
+    @property
+    def invoiced_pricing(self) -> Pricing | None:
+        processed_pricings = [pricing for pricing in self.pricings if pricing.status == PricingStatus.INVOICED]
+
+        if not processed_pricings:
+            return None
+
+        pricings = sorted(processed_pricings, key=attrgetter("creationDate"), reverse=True)
+
+        if pricings:
+            return pricings[0]
+
+        return None
+
+    @property
+    def regulation_cashflow_batch(self) -> CashflowBatch | None:
+        if not self.pricing:
+            return None
+
+        cashflow = self.pricing.cashflow
+        if not cashflow:
+            return None
+
+        return cashflow.batch
+
+
+class BookingFinanceIncident(Base, Model):
+    id: int = sqla.Column(sqla.BigInteger, primary_key=True, autoincrement=True)
+
+    bookingId = sqla.Column(sqla.BigInteger, sqla.ForeignKey("booking.id"), index=True, nullable=True)
+    booking: sqla_orm.Mapped["bookings_models.Booking | None"] = sqla_orm.relationship(
+        "Booking", foreign_keys=[bookingId]
+    )
+
+    collectiveBookingId = sqla.Column(
+        sqla.BigInteger, sqla.ForeignKey("collective_booking.id"), index=True, nullable=True
+    )
+    collectiveBooking: sqla_orm.Mapped["educational_models.CollectiveBooking | None"] = sqla_orm.relationship(
+        "CollectiveBooking",
+        foreign_keys=[collectiveBookingId],
+    )
```

Comment:
Dans une **autre** PR (pour éviter de retarder encore le merge de celle-ci, et car ça n'est pas bloquant), il faudrait ajouter une contrainte qui vérifie qu'on a soit un `bookingId`, soit un `collectiveBookingId` (mais pas les deux, ni aucun des deux). Quelque chose de similaire à ça : https://github.com/pass-culture/pass-culture-main/blob/4bbfb368447cd6582ab5f0536f7281571adf0dcc/api/src/pcapi/core/finance/models.py#L246-L247

Reply:
C'est noté :slightly_smiling_face: